autonomy/chain/tx.py

```diff
@@ -131,6 +137,23 @@ def transact(
             except Exception as e:  # pylint: disable=broad-except
                 if not should_retry(str(e)):
                     raise ChainInteractionError(str(e)) from e
+                if should_reprice(str(e)):
+                    print("Repricing the transaction...")
+                    tx_dict = cast(Dict, tx_dict)
+                    old_price = {
+                        "maxFeePerGas": tx_dict[  # pylint: disable=unsubscriptable-object
+                            "maxFeePerGas"
+                        ],
+                        "maxPriorityFeePerGas": tx_dict[  # pylint: disable=unsubscriptable-object
+                            "maxPriorityFeePerGas"
+                        ],
+                    }
```

Comment:
This is problematic. You tie the implementation to Ethereum plugin. Repricing should happen behind a generic api and be handled on the plugin. So that the CLI doesn't need to know about the internals.